main.js

```diff
@@ -0,0 +1,141 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
+const boxes = document.querySelectorAll('.box');
+
+const Player = (name, symbol) => ({ name, symbol });
+const player1 = Player('Player 1', 'X');
+const player2 = Player('Player 2', 'O');
+
+const gameBoard = (() => {
+  let board = new Array(9).fill(0);
+
+  const disable = () => {
+    boxes.forEach((box) => {
+      box.disabled = true;
+    });
+  };
+
+  const restart = () => {
+    boxes.forEach((box) => {
+      box.disabled = false;
+      box.value = '';
+    });
+    board = new Array(9).fill(0);
+  };
+
+  const turn = () => {
+    const zero = board.filter(a => a === 0);
+    if (zero.length % 2 === 0) {
+      return player2;
+    }
+    return player1;
+  };
+
+  const isFull = () => {
+    const zero2 = board.filter(a => a === 0);
+    if (zero2.length === 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const checkWinner = () => {
+    const combos = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [2, 4, 6],
+      [0, 4, 8],
+    ];
+
+    let winner = false;
+    combos.forEach((combo) => {
```

Comment:
To avoid having to reassign `winner` here I suggest we just have a nested `every` inside a `some` method. This will automatically return either true or false for us. We can then just return that call to `some` (which inside it nests an `every` call) as the main return to this `checkWinner` function.

Reply:
Can you please give us a hint about this one. We would love to optimize the way we look for the winning combo, but we don't understand how  to use `some` here, because `some` is called on an array and if call it on `board` or `combo` it is going to always return true. This is what we tried:
```
function checkWinner(params){
  combos.forEach((combo) => {
    (combo.every(e => board[e] === 'X') || combo.every(e => board[e] === 'O')
  });
}  
console.log(combos.some(checkWinner))
```

Reply:
See my comment above as reply to this :)

Reply:
Solved! Thanks for your help relating to some method